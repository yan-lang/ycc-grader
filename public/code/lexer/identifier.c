apple
_apple
__apple
____appple___
_apple_apple_
a_pple
a__apple
apple123
int1234
_int
_i12345
in12344t