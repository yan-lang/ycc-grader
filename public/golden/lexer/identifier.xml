<tokens>
    <token>
        <index>0</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>apple</value>
        <line>1</line>
        <column>1</column>
        <start>0</start>
        <stop>5</stop>
    </token>
    <token>
        <index>1</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>_apple</value>
        <line>2</line>
        <column>1</column>
        <start>6</start>
        <stop>12</stop>
    </token>
    <token>
        <index>2</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>__apple</value>
        <line>3</line>
        <column>1</column>
        <start>13</start>
        <stop>20</stop>
    </token>
    <token>
        <index>3</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>____appple___</value>
        <line>4</line>
        <column>1</column>
        <start>21</start>
        <stop>34</stop>
    </token>
    <token>
        <index>4</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>_apple_apple_</value>
        <line>5</line>
        <column>1</column>
        <start>35</start>
        <stop>48</stop>
    </token>
    <token>
        <index>5</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>a_pple</value>
        <line>6</line>
        <column>1</column>
        <start>49</start>
        <stop>55</stop>
    </token>
    <token>
        <index>6</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>a__apple</value>
        <line>7</line>
        <column>1</column>
        <start>56</start>
        <stop>64</stop>
    </token>
    <token>
        <index>7</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>apple123</value>
        <line>8</line>
        <column>1</column>
        <start>65</start>
        <stop>73</stop>
    </token>
    <token>
        <index>8</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>int1234</value>
        <line>9</line>
        <column>1</column>
        <start>74</start>
        <stop>81</stop>
    </token>
    <token>
        <index>9</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>_int</value>
        <line>10</line>
        <column>1</column>
        <start>82</start>
        <stop>86</stop>
    </token>
    <token>
        <index>10</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>_i12345</value>
        <line>11</line>
        <column>1</column>
        <start>87</start>
        <stop>94</stop>
    </token>
    <token>
        <index>11</index>
        <type>IDENTIFIER</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>in12344t</value>
        <line>12</line>
        <column>1</column>
        <start>95</start>
        <stop>103</stop>
    </token>
    <token>
        <index>12</index>
        <type>EOF</type>
        <source>identifier.c</source>
        <lexer>Lexer</lexer>
        <value>null</value>
        <line>12</line>
        <column>9</column>
        <start>103</start>
        <stop>103</stop>
    </token>
</tokens>